contextmanager.go

```diff
@@ -59,6 +59,22 @@ func newContextManager(label string) *contextManager {
 	return c
 }
 
+// Pick the first matching transfer syntax UID appeares in allTransferSyntaxes
+// - If one of the client proposed transfer syntaxes matches the server's preferred transfer syntax, then that transfer syntax is accepted.
+// - If the client does not propose a transfer syntax that matches the server's preferred transfer syntax, the first transfer syntax in the client's list of proposed syntaxes is accepted.
+// - If none of the proposed transfer syntaxes are supported, the server MUST send a bind_ack with all transfer syntaxes rejected.
+func pickFirstPreferedTransferSyntax(proposedTransferSyntaxUIDs []string) string {
+	for _, syntaxUID := range allTransferSyntaxes {
+		for _, proposed := range proposedTransferSyntaxUIDs {
+			if syntaxUID == proposed {
+				return syntaxUID
+			}
+		}
+	}
```

Comment:
Could you use debugger to see if what the client proposed transfer syntaxes looks like? Are they sorted by preference? Because if not, and the client proposed transfer syntaxes list look like

```
["Implicit VR Little Endian", "JPEG Baseline", ...]
```

the default `"Implicit VR Little Endian"` will be picked every time.

Reply:
Here is what they proposed:
```
D: Presentation Contexts:
D:   Context ID:        1 (Proposed)
D:     Abstract Syntax: =ComputedRadiographyImageStorage
D:     Proposed SCP/SCU Role: SCU
D:     Proposed Transfer Syntax(es):
D:       =JPEG2000
D:       =LittleEndianExplicit
D:       =LittleEndianImplicit
D:   Context ID:        3 (Proposed)
D:     Abstract Syntax: =SecondaryCaptureImageStorage
D:     Proposed SCP/SCU Role: SCU
D:     Proposed Transfer Syntax(es):
D:       =JPEG2000
D:       =LittleEndianExplicit
D:       =LittleEndianImplicit
```
> Are they sorted by preference?

Yes